src/ophyd_async/epics/areadetector/writers/general_hdffile.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Iterator, List, Optional
+from urllib.parse import urlunparse
+
+import event_model
+from event_model import (
+    ComposeStreamResource,
+    ComposeStreamResourceBundle,
+    StreamDatum,
+    StreamResource,
+)
+
+from ophyd_async.core import DirectoryInfo
+
+
+@dataclass
+class _HDFDataset:
+    name: str
+    dtype_numpy: Optional[str] = None
+    swmr: bool = False
+    shape: Optional[List[int]] = None
+    multiplier: Optional[int] = 1
+    path: Optional[str] = None
```

Comment:
And this one has also confused us in the past, I suggest:
```suggestion
    #: Name of the dataset within the HDF file
    dataset: str
```